tests/postgres_tests/test_aggregates.py

```diff
@@ -183,6 +183,25 @@ def test_string_agg_charfield_ordering(self):
                 )
                 self.assertEqual(values, {'stringagg': expected_output})
 
+    def test_string_agg_charfield_ordering_subquery(self):
+        subquery = AggregateTestModel.objects.annotate(stringagg=StringAgg(
+            'char_field',
+            delimiter=';',
+            ordering='-char_field',
+        )).exclude(char_field=OuterRef('char_field')).values('stringagg')
+        subquery.query.set_group_by()
```

Comment:
Tests shouldn't rely on internal APIs to trigger a bug.

Reply:
@charettes the `set_group_by` is not being used to trigger the bug, but to make the subquery work due to issue described in [Ticket #29296](https://code.djangoproject.com/ticket/28296)

Reply:
@caioariede I assume you are referring to [this comment](https://code.djangoproject.com/ticket/28296#comment:2)? In all cases the crash should be reproduced without relying on internal APIs in tests. If you want to avoid a `NULL` to be returned use `Coalesce`.

Reply:
@charettes no, the problem is that without `set_group_by()` the subquery will have a `GROUP BY "id"` clause that is incorrect:

```sql
SELECT "postgres_tests_aggregatetestmodel"."char_field",
(
  SELECT STRING_AGG(U0."char_field", ';'ORDER BY U0."char_field" DESC) AS "stringagg"
  FROM "postgres_tests_aggregatetestmodel" U0
  WHERE NOT (U0."char_field" = "postgres_tests_aggregatetestmodel"."char_field")
  GROUP BY U0."id" -- HERE
)
AS "others" FROM "postgres_tests_aggregatetestmodel" ORDER BY "postgres_tests_aggregatetestmodel"."char_field" ASC
```

and **with** the `set_group_by` it generates the correct SQL:

```sql
SELECT "postgres_tests_aggregatetestmodel"."char_field",
(
  SELECT STRING_AGG(U0."char_field", ';'ORDER BY U0."char_field" DESC) AS "stringagg" 
  FROM "postgres_tests_aggregatetestmodel" U0
  WHERE NOT (U0."char_field" = "postgres_tests_aggregatetestmodel"."char_field")
)
AS "others" FROM "postgres_tests_aggregatetestmodel" ORDER BY "postgres_tests_aggregatetestmodel"."char_field" ASC
```

Reply:
@caioariede I understand, that's the exact issue the comment I linked to describes. You still don't want to be using an internal API to test this out; the ORM doesn't officially support aggregation through subqueries and `subquery.query.group_by` alteration is a hack we don't want to rely on in regressions tests. In summary we don't support this pattern and we shouldn't test for it.

I still think that if we should focus on the crash here and the fact the reported ticket joins a relation where your test doesn't (`StringAgg('people__first_name')`) which is why it requires the `set_group_by` to return the appropriate results.

Reply:
I understand it now, thanks @charettes. I'll provide the changes.

Reply:
I've updated it @charettes